circus_tower_problem/circus_tower.cpp

```diff
@@ -0,0 +1,89 @@
+#include "circus_tower.h"
+
+void BinaryInsert(size_t person, size_t i, size_t j,
```

Comment:
this algorithm is pretty complex, which makes it hard for me to judge whether it is correct. 
If you want to polish this task (instead of e.g. moving to another one), 
I suggest that you implement a worse-case-quadratic algorithm and compare the results between the two algorithms on large auto-generated inputs 

outline: 
1. sort by h-then-w (or vise versa?)
2. for each person, scan the array upward and add whoever satisfies the rules